pilot/pkg/xds/workload.go

```diff
@@ -151,18 +152,27 @@ func (e WorkloadRBACGenerator) GenerateDeltas(
 	var updatedPolicies sets.Set[model.ConfigKey]
 	if len(req.ConfigsUpdated) != 0 {
 		updatedPolicies = model.ConfigsOfKind(req.ConfigsUpdated, kind.AuthorizationPolicy)
-		updatedPolicies = updatedPolicies.Merge(model.ConfigsOfKind(req.ConfigsUpdated, kind.PeerAuthentication))
+		// Convert the actual Kubernetes PeerAuthentication policies to the synthetic ones
+		// by adding the prefix
+		for p := range model.ConfigsOfKind(req.ConfigsUpdated, kind.PeerAuthentication) {
+			updatedPolicies.Insert(model.ConfigKey{
+				Name:      controller.GetAmbientPolicyConfigName(p),
+				Namespace: p.Namespace,
+				Kind:      p.Kind,
+			})
+		}
 	}
 	if len(req.ConfigsUpdated) != 0 && len(updatedPolicies) == 0 {
 		// This was a incremental push for a resource we don't watch... skip
 		return nil, nil, model.DefaultXdsLogDetails, false, nil
 	}
+
 	policies := e.s.Env.ServiceDiscovery.Policies(updatedPolicies)
```

Comment:
seems like the "convert to syntethic ones" should be handled in Policies()? otherwise we are kind of leaking a bit

Reply:
maybe not actually since we need to know them for the expected/removed stuff as well..

Reply:
Exactly, that was the crux of the issue. The diffing was off so ztunnel removed the policy from its local state, leading to the policies being ignored

Reply:
The real root cause of the leaking abstraction is the ConfigUpdate being pretty lose with what events we send. We do the same in other places though, not so bad just a bit tricky for cases like this.

Reply:
Yeah it looks like PeerAuthentication gets handled in the generic `configHandler` [here](https://github.com/istio/istio/blob/master/pilot/pkg/bootstrap/server.go#L882). I can add a TODO to split PeerAuthentication out with its own logic if you think it'll be helpful